Improve VS binlogs providing info (#8446)
dotnet/msbuild: documentation/wiki/Providing-Binary-Logs.md

⚠ NOTE: some build environments make secrets available using environment variables. Before sharing a binary log, make sure it does not expose API tokens or other important secrets.

You can create a binary log by passing the `-bl` parameter to MSBuild (`MSBuild.exe` or `dotnet build`). You can explore the contents of the generated .binlog file using [MSBuild Structured Log Viewer](http://msbuildlog.com/) or in your browser using [Live Structured Log Viewer](https://live.msbuildlog.com). Note: We don't capture any data from binary logs viewed on your browser.

[More details about binary logs](Binary-Log.md)

## Capturing Binary Logs Through Visual Studio
### (Preferred way) Capturing logs for all MSBuild invocations
Via setting `MSBUILDDEBUGENGINE` environment variable to `'1'`:
```
> SET MSBUILDDEBUGENGINE=1
> devenv.exe MySolution.sln
```

> MSBuild binary logs are then captured to `MSBuild_Logs` subfolder of a current folder.

Reviewer: I do not find this reliable and generally prefer explicitly specifying `MSBUILDDEBUGPATH`.

Reviewer 2: (If MSBUILDDEBUGPATH is unset, it tries the current folder then temp...it's more confusing.)

⚠ NOTE: logs are being recorded for each MSBuild invocation (including design time builds) and kept in the folder without removing older ones - so the number of log files can grow quickly. It is recommended to set the opt-in environment variable only for the short duration of reproducing the issue to be investigated (though it is understandable that some nondeterministic issues might need multiple reproduction attempts)

> [More technical info](Building-Testing-and-Debugging-on-Full-Framework-MSBuild.md#logs)

Reviewer: Maybe also worth crosslinking to https://github.com/dotnet/project-system/blob/3742e229b41a086d738745a5e5a7a15f535deff7/docs/repo/debugging/design-time-builds.md#gathering-full-fidelity-binlogs

⚠ NOTE: some build environments make secrets available using environment variables. Before sharing a binary log, make sure it does not expose API tokens or other important secrets.

You can create a binary log by passing the `-bl` parameter to MSBuild (`MSBuild.exe` or `dotnet build`). You can explore the contents of the generated .binlog file using [MSBuild Structured Log Viewer](http://msbuildlog.com/) or in your browser using [Live Structured Log Viewer](https://live.msbuildlog.com). Note: We don't capture any data from binary logs viewed on your browser.

[More details about binary logs](Binary-Log.md)

## Capturing Binary Logs Through Visual Studio
### (Preferred way) Capturing logs for all MSBuild invocations

> Via setting `MSBUILDDEBUGENGINE` environment variable to `'1'`:

Reviewer 2: I'm not sure if this is obvious to senior developers, but I had absolutely no idea how to set environment variables that affected Visual Studio before I was told to set the environment variables in a command prompt then open VS from that command prompt. That probably bears mentioning here.

Author: Very good feedback - since we are linking this in external items (asking customers for more data) - it should be super clear - updating..

```
> SET MSBUILDDEBUGENGINE=1
> devenv.exe MySolution.sln
```

> MSBuild binary logs are then captured to a location specified via `MSBUILDDEBUGPATH` environment variable (or defaults to `MSBuild_Logs` subfolder of a current folder or `%temp%`, based on access rights).

Reviewer 2: nit:
I'd add "and setting `MSBUILDDEBUGPATH` to a folder you can find" to line 15 above, then noting the defaults if you don't set it here. The idea is that we'd rather they just set that path and not rely on the defaults...but if it's not set, the feature still works.

⚠ NOTE: logs are being recorded for each MSBuild invocation (including design time builds) and kept in the folder without removing older ones - so the number of log files can grow quickly. It is recommended to set the opt-in environment variable only for the short duration of reproducing the issue to be investigated (though it is understandable that some nondeterministic issues might need multiple reproduction attempts)

Further reading:
* [More technical info](Building-Testing-and-Debugging-on-Full-Framework-MSBuild.md#logs)

> * [Design time builds logs](https://github.com/dotnet/project-system/blob/main/docs/repo/debugging/design-time-builds.md#gathering-full-fidelity-binlogs)

Reviewer 2: ...and added a bug on the PS repo https://github.com/dotnet/project-system/issues/8857

😇

Author: Perfect!

Do you want to shoot a quick PR to their docs to rectify it right away?

Reviewer 2: MiYanni said he didn't want it fixed, sadly.

⚠ NOTE: some build environments make secrets available using environment variables. Before sharing a binary log, make sure it does not expose API tokens or other important secrets.

You can create a binary log by passing the `-bl` parameter to MSBuild (`MSBuild.exe` or `dotnet build`). You can explore the contents of the generated .binlog file using [MSBuild Structured Log Viewer](http://msbuildlog.com/) or in your browser using [Live Structured Log Viewer](https://live.msbuildlog.com). Note: We don't capture any data from binary logs viewed on your browser.

[More details about binary logs](Binary-Log.md)

## Capturing Binary Logs Through Visual Studio
### (Preferred way) Capturing logs for all MSBuild invocations

> Via setting `MSBUILDDEBUGENGINE` environment variable to `'1'` and (optionaly) `MSBUILDDEBUGPATH` to an existing destination folder to store the captured logs and starting the Visual Studio from the same shell to inherit the environment:

Reviewer 2: A short fragment feels fine to me, but this is getting a little unwieldy.

```suggestion
Set `MSBUILDDEBUGENGINE` environment variable to `'1'` and (optionally) set `MSBUILDDEBUGPATH` to an existing destination folder to store the captured logs. Then start Visual Studio from the same shell to inherit the environment:
```
